laurent22/joplin
# Mobile: Plugin support: Simplify reporting plugin issues

# Summary

This pull request is part of a follow-up to the "show plugin description banners" pull request (see [comment](https://github.com/laurent22/joplin/pull/10286#discussion_r1561805855)). It does the following:
- Adds an "about" screen for each plugin.
    - To support this, refactors the logic for displaying a modal with a close button out of `PluginPanelViewer`.
- Includes information about how to report bugs in a plugin on that about screen.

**NOTES**:
- Currently, the "report on Joplin forum" option auto-fills a post in the "Support" category (see [screenshot](https://github.com/laurent22/joplin/pull/10319#discussion_r1567989998) below for details).
- This pull request is being opened in part as to discuss UI design for this feature. Feedback is welcome :slightly_smiling_face: .

# Screenshots

![A grid of screenshots: Includes light and darkmode screenshots of two plugins: Outline and create and goto at tags and pound notebooks. Each has "About" and "report fraudulent plugin" links. Only outline has a "report an issue" link](https://github.com/laurent22/joplin/assets/46334387/b0dda52e-7835-46b6-a100-12c49a63b229)




# Testing

1. Install 2 or more plugins (`outline` and `create and goto #tags and @notebooks`)
2. Verify that installed plugins have an "info" button on the right of their titles.
3. Verify that clicking the "info" button (test for `outline`) shows a screen with the plugin's title, author, description, and link buttons.
4. Verify that clicking "About" opens the plugin website page for a plugin.
5. Open a plugin with a GitHub repository's info screen (e.g. the outline plugin). Verify that clicking "report an issue" opens the GitHub issues page for `outline`.
6. Verify that clicking "report fraudulent plugin" opens Joplin's page for reporting security advisories.

This has been tested successfully on a physical Android device. An earlier version of this pull request was also tested on an iOS simulator.
